### PR TITLE
Bugfix/rlp 854/invalid close signature

### PR DESCRIPTION
**Pull request by mauricioirace**
TODO: Add notes and changes

### CODE REVIEW

File: raiden/api/webui/static/endpointConfig.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-const backendUrl='http://localhost:5012'; 
-const nodeAddress = '0x55825d1DA0576780926568e208f7D3F55B89Cf1a'; 
+const backendUrl='http://localhost:5002';
```
Comment (mauricioirace) [inline raiden/api/webui/static/endpointConfig.js]: Again...

File: raiden/blockchain_events_handler.py [hunk 2]
```diff
@@ -171,6 +133,77 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
...
+                                   token_network_identifier,
+                                   participant1: ChecksumAddress,
+                                   participant2: ChecksumAddress,
+                                   raiden):
+    is_participant1_handled_lc = LightClientService.is_handled_lc(to_checksum_address(encode_hex(participant1)),
+                                                                  raiden.wal)
+    is_participant2_handled_lc = LightClientService.is_handled_lc(to_checksum_address(encode_hex(participant2)),
+                                                                  raiden.wal)
+    is_light_channel = is_participant1_handled_lc or is_participant2_handled_lc
+    both_participants_are_light_clients = is_participant1_handled_lc and is_participant2_handled_lc
```
Comment (mauricioirace) [inline raiden/blockchain_events_handler.py]: Here we are checking for second time if particpant is lc. This could be improved

File: raiden/api/webui/static/endpointConfig.js (hunk 1 above)
Comment (marcosmartinez7) [reply raiden/api/webui/static/endpointConfig.js]: not again decia

**Comment by marcosmartinez7** [comment]
It would be nice a quick call to present this PR :)

File: raiden/transfer/token_network.py [hunk 3]
```diff
@@ -327,14 +319,6 @@ def state_transition(
         iteration = handle_channel_close(
             token_network_state, state_change, block_number, block_hash
         )
-    elif type(state_change) == ActionChannelSetFee:
```
Comment (JONAF2103) [inline raiden/transfer/token_network.py:332]: Why is this removed, we remove the Channel set fee action? maybe i miss understood the point of the change

File: raiden/api/python.py [hunk 4]
```diff
@@ -624,11 +624,10 @@ def set_total_channel_deposit_light(
         )
 
         channel_proxy = self.raiden.chain.payment_channel(
+            creator_address,
             canonical_identifier=channel_state.canonical_identifier
         )
 
-        channel_proxy.swap_participants(creator_address)
```
Comment (marcosmartinez7) [inline raiden/api/python.py:705]: why this was removed?

File: raiden/blockchain_events_handler.py [hunk 5]
```diff
@@ -171,6 +133,77 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
...
+    participant2,
+    token_network_identifier,
+    transaction_hash,
+    raiden):
+    channel_state = create_channel(block_hash, block_number, channel_identifier, participant1, participant2, raiden,
+                                   token_network_identifier, transaction_hash)
+
+    partner_address = channel_state.partner_state.address
+
+    if ConnectionManager.BOOTSTRAP_ADDR != partner_address:
```
Comment (marcosmartinez7) [inline raiden/blockchain_events_handler.py]: this should be at create_channel

File: raiden/api/webui/static/endpointConfig.js (hunk 1 above)
Comment (mortelli) [reply raiden/api/webui/static/endpointConfig.js]: please undo the changes to this file

File: raiden/network/blockchain_service.py [hunk 6]
```diff
@@ -212,7 +213,7 @@ def service_registry(self, address: Address) -> ServiceRegistry:
 
         return self.address_to_service_registry[address]
 
-    def payment_channel(self, canonical_identifier: CanonicalIdentifier) -> PaymentChannel:
+    def payment_channel(self, participant1: Address, canonical_identifier: CanonicalIdentifier) -> PaymentChannel:
```
Comment (mortelli) [inline raiden/network/blockchain_service.py]: i don't think it makes much sense to call this `participant1` if it's the only participant parameter in this signature.

do we have any alternatives to this name? `creator_address`?

File: raiden/transfer/channel.py [hunk 7]
```diff
@@ -1561,11 +1562,12 @@ def handle_action_close(
...
-    events = events_for_close(
-        channel_state=channel_state, block_number=block_number, block_hash=block_hash,
-        signed_close_tx=close.signed_close_tx
-    )
+    events = []
+    if close.participant1 == channel_state.our_state.address:
+        events = events_for_close(
+            channel_state=channel_state, block_number=block_number, block_hash=block_hash,
+            signed_close_tx=close.signed_close_tx
+        )
```
Comment (mortelli) [inline raiden/transfer/channel.py:1571]: can you please explain the reason behind this change?

why do we not care about `close.participant2`, for example?

File: raiden/network/blockchain_service.py [hunk 8]
```diff
@@ -224,17 +225,20 @@ def payment_channel(self, canonical_identifier: CanonicalIdentifier) -> PaymentC
 
         with self._payment_channel_creation_lock:
             dict_key = (token_network_address, channel_id)
```
Comment (mortelli) [inline raiden/network/blockchain_service.py]: let's please rename this to something meaningful, e.g. `channel_identifier`

File: raiden/blockchain_events_handler.py [hunk 9]
```diff
@@ -87,67 +88,28 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
...
-            light_client_address = participant2
-
-        if ConnectionManager.BOOTSTRAP_ADDR != partner_address:
-            raiden.start_health_check_for(partner_address, light_client_address)
-
+            create_light_channel(block_hash, block_number, channel_identifier,
+                                 participant1, participant2, token_network_identifier,
+                                 transaction_hash, raiden)
+        if is_participant2_handled_lc:
+            create_light_channel(block_hash, block_number, channel_identifier,
```
Comment (mortelli) [inline raiden/blockchain_events_handler.py]: be careful, this can create 2 channels. is this intended?

File: raiden/blockchain_events_handler.py [hunk 10]
```diff
@@ -87,67 +88,28 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
...
-            channel_state=channel_state,
-            block_number=block_number,
-            block_hash=block_hash,
-        )
-        raiden.handle_and_track_state_change(new_channel)
-
-        partner_address = channel_state.partner_state.address
-
-        light_client_address = None
+    if is_light_channel:
```
Comment (mortelli) [inline raiden/blockchain_events_handler.py:107]: this variable serves no purpose. just do:

```python
if is_p1_lc:
  create_lc(...)
if is_p2_lc:  # should this be elif instead?
  create_lc(...)
elif:
  ...
```

File: raiden/blockchain_events_handler.py [hunk 11]
```diff
@@ -87,67 +88,28 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
...
+        if is_participant2_handled_lc:
+            create_light_channel(block_hash, block_number, channel_identifier,
+                                 participant2, participant1, token_network_identifier,
+                                 transaction_hash, raiden)
+    elif raiden.address == participant1:
+        create_channel(block_hash, block_number, channel_identifier, participant1, participant2, raiden,
+                       token_network_identifier, transaction_hash)
+    elif raiden.address == participant2:
+        create_channel(block_hash, block_number, channel_identifier, participant2, participant1, raiden,
+                       token_network_identifier, transaction_hash)
```
Comment (mortelli) [inline raiden/blockchain_events_handler.py]: i would rather you defined the participant variables outside (one way or the other, depending on `raiden.address`) and then just have 1 call to `create_channel`. 

for me, it's much easier to read and maintain this way

Comment (mortelli) [reply raiden/blockchain_events_handler.py]: (same with the `create_light_channel` calls)

File: raiden/blockchain_events_handler.py (hunk 9 above)
Comment (mortelli) [reply raiden/blockchain_events_handler.py]: (i can suggest another code structure for this method depending on the answer to this question)

File: raiden/blockchain_events_handler.py (hunk 5 above)
Comment (mortelli) [reply raiden/blockchain_events_handler.py]: if this is true, then i don't know if there is much point in having this function instead of calling `create_channel` directly

File: raiden/blockchain_events_handler.py (hunk 2 above)
Comment (mortelli) [reply raiden/blockchain_events_handler.py]: you can use this `and` as an in-line assignment on line `202` instead

Comment (mortelli) [reply raiden/blockchain_events_handler.py]: the same applies to the variable set on `182`

File: raiden/blockchain_events_handler.py [hunk 12]
```diff
@@ -171,6 +133,77 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
...
+        token_address=typing.TokenAddress(token_address),
+        payment_network_identifier=raiden.default_registry.address,
+        token_network_address=token_network_identifier,
+        reveal_timeout=raiden.config["reveal_timeout"],
+        payment_channel_proxy=channel_proxy,
+        opened_block_number=block_number,
+        is_light_channel=is_light_channel,
+        both_participants_are_light_clients=both_participants_are_light_clients
+    )
+    return channel_state
```
Comment (mortelli) [inline raiden/blockchain_events_handler.py]: you can just call `return` on line `194`

**Review by mortelli — COMMENTED** [review]
heads-up: there's a conflict in `raiden/transfer/token_network.py`

File: raiden/blockchain_events_handler.py (hunk 10 above)
Comment (mauricioirace) [reply raiden/blockchain_events_handler.py:107]: It's ok as an if, as both could be light clients.

I don't think the code you suggest is actually equivalent to the original one.

File: raiden/blockchain_events_handler.py (hunk 9 above)
Comment (mauricioirace) [reply raiden/blockchain_events_handler.py]: yes, It's intended: it's a workaround taken from the base branch. We duplicate the channels in case both participants are lc, but with the participants exchanged

File: raiden/blockchain_events_handler.py (hunk 10 above)
Comment (mauricioirace) [reply raiden/blockchain_events_handler.py:107]: What about something like:

```suggestion
    if is_light_channel or raiden.address in (participant1, participant2):
        if raiden.address == participant1 or is_participant1_handled_lc:
            create_channel(
                block_hash,
                block_number,
                channel_identifier,
                participant1,
                participant2,
                token_network_identifier,
                transaction_hash,
                raiden
            )
        if raiden.address == participant2 or is_participant2_handled_lc:
            create_channel(
                block_hash,
                block_number,
                channel_identifier,
                participant2,
                participant1,
                token_network_identifier,
                transaction_hash,
                raiden)
    # Raiden node is not participant of channel. Lc are not participants
    else:
```

File: raiden/transfer/token_network.py (hunk 3 above)
Comment (mauricioirace) [reply raiden/transfer/token_network.py:332]: yes, I mean, it never existed

File: raiden/transfer/channel.py (hunk 7 above)
Comment (mauricioirace) [reply raiden/transfer/channel.py:1571]: Now (since banche RLP-715) we have two channels per channel (one per side), but we only want to close it once, or else, it will fail. Also, the `close.signed_close_tx` will not verify for both sides of the channel.
